contrib/src/main/scala/akka/stream/contrib/PassThroughFlow.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.contrib
+
+import akka.NotUsed
+import akka.stream._
+import akka.stream.scaladsl._
+
+/**
+ * Given `Flow[In, Out]` and `f: (In, Out) => Result`,
+ * transforms original flow into a `Flow[In, Result]`, i.e.
+ * for every element `e` signalled from upstream, emits `f(e, flow(e))`.
```

Comment:
Very important that the flow is 1:1, and does no re-ordering, as soon as it drops, creates more elements or re-orders the downstream be incorrect. (Which is the reason there is no built in tool for it, we have discussed it over and over again to try to figure out solutions though, see https://github.com/akka/akka/issues/15957 and linked issues)

Reply:
Right, that's what I also gathered from the links in the original discussion. Do you want me to make this more explicit in docs or do you mean that since it's not a generic solution, it shouldn't be here?

Reply:
I think it needs to be a *blink UPPERCASE* warning in the top of the docs but ok in contrib if that is added.

Reply:
done